tests/framework/interface.go

```diff
@@ -30,7 +30,7 @@ type testRunner interface {
 
 type Cluster interface {
 	Members() []Member
-	Client() Client
+	Client(cfg clientv3.AuthConfig) (Client, error)
```

Comment:
I am thinking whether it's possible to get rid of the parameter `cfg clientv3.AuthConfig`. When the test cases setup the user & role, and enable the auth, then the client should record the authConfig automatically. 

Does this make sense?

Reply:
I don't think client should be stateful. In general, it makes the client not concurrent safe. If multiple callers set up the user & role and enable the auth using the client, this would cause contention.

Reply:
ack. 

It's just a little strange to provide a specific` cfg clientv3.AuthConfig` parameter, but automatically get all other parameters from the cluster.  Probably it makes more sense to resolve this using a generic way, such as change the parameter to something like `opts ...Option`.  It might be a little over engineering from another perspective. 

Let's keep it as it's for now, and probably think about it separately.